
# HG changeset patch
# User Michael Weghorn <m.weghorn@posteo.de>
# Date 1761894457 -7200
# Node ID 5a81c1541639111d2b525d83f5f35e9715d1c4cf
# Parent  cdaca1aa99d2d68db98c1a30173c4e87585ceb5e
Bug 1901089: Disable date pickers when their checkboxes are unchecked. r=darktrojan


The updateDateCheckboxes function contains logic
supposed to disable the date/time pickers when the
related checkboxes are unchecked.

However, that logic wasn't working because
it tried to set a property directly on the ID of
the corresponding date/time picker element instead
of on the actual element.

Make this actually work by retrieving the
corresponding element and setting its
"disabled" property.

Also adjust logic to set the value a few lines
above to also use the element instead of its ID.

Differential Revision: https://phabricator.services.mozilla.com/D269683

diff --git a/calendar/base/content/item-editing/calendar-item-iframe.js b/calendar/base/content/item-editing/calendar-item-iframe.js
--- a/calendar/base/content/item-editing/calendar-item-iframe.js
+++ b/calendar/base/content/item-editing/calendar-item-iframe.js
@@ -1281,21 +1281,23 @@ function updateDateCheckboxes(aDatePicke
     return;
   }
 
   if (!window.calendarItem.isTodo()) {
     return;
   }
 
   // force something to get set if there was nothing there before
-  aDatePickerId.value = document.getElementById(aDatePickerId).value;
+  const datePicker = document.getElementById(aDatePickerId);
+  // eslint-disable-next-line no-self-assign
+  datePicker.value = datePicker.value; // FIXME: check if really needed
 
   // first of all disable the datetime picker if we don't have a date
   const hasDate = document.getElementById(aCheckboxId).checked;
-  aDatePickerId.disabled = !hasDate;
+  datePicker.setAttribute("disabled", !hasDate);
 
   // create a new datetime object if date is now checked for the first time
   if (hasDate && !aDateTime.isValid()) {
     const date = cal.dtz.jsDateToDateTime(
       document.getElementById(aDatePickerId).value,
       cal.dtz.defaultTimezone
     );
     aDateTime.setDateTime(date);

